code-challenges.test.js

```diff
@@ -23,27 +43,132 @@ const secretCodeWord3 = "Eccentric"
+//         console.log(stringToArr[i])
+//         if (stringToArr[i] === "a" || stringToArr[i] === "A") { 
+//         // reassign the value at the index to 4
+//             stringToArr[i] = 4
+//         } else if (stringToArr[i] === "e" || stringToArr[i] === "E") {
+//             stringToArr[i] = 3
+//         } else if (stringToArr[i] === "i" || stringToArr[i] === "I") {
+//             stringToArr[i] = 1
+//         } else if (stringToArr[i] === "o" || stringToArr[i] === "O") {
+//             stringToArr[i] = 0
+//         } else {
+//             "whoopsie"
+//         }
+//     }
+//     return stringToArr.join("")
+// }
+
+// console.log(codedMessage(secretCodeWord2))
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
+// describe("particularLetter", () => {
+//     it("returns an array of all the words containing that particular letter", () => {
+//         const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+//         expect(particularLetter(filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+//         expect(particularLetter(filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+//     })
+// })
+
 const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
 
+// FAIL  ./code-challenges.test.js
+// particularLetter
+//   ✕ returns an array of all the words containing that particular letter (1 ms)
+// ● particularLetter › returns an array of all the words containing that particular letter
+//   ReferenceError: particularLetter is not defined
+
+
 const filterLetterA = "a"
 // Expected output: ["Mango", "Apricot", "Peach"]
 const filterLetterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
 
+// Create a function that takes in an array of words and a single letter and returns an array of
+// all the words containing that particular letter.
+
 // Pseudo code:
+// input: array of words ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+// output: an array of all the words containing that particular letter
+//         'a' ["Mango", "Apricot", "Peach"] 
+//         'e' ["Cherry", "Blueberry", "Peach"]
+// process: will need to iterate over array to return requested words
+//          .filter immediately comes to mind, though not as good with this one 
+//          could use for loop?
+//          will need to include looking for capital letters as well, 
+//          keyword 'if' so maybe conditional instead
+
+const particularLetter = (array) => {
+    // let newArray = []
+    for(let i = 0; i < array.length; i++) {
+        // I dont think && 'e' || 'E' is correct because im telling it to look for both at once
+        if(array[i].includes('a' || 'A' && 'e' || 'E')){
```

Comment:
This is a solid attempt, in you refactor really look at your prompt and see what the function is taking in and also think about using the empty array you commented out. What can you do to on each iteration with your empty array?